js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -1655,130 +1680,151 @@ const EditingController = modules.ViewController.inherit((function() {
                 }
             });
         },
+
         saveEditData: function() {
             const deferred = new Deferred();
-            const afterSaveEditData = (cancel) => {
-                when(this._afterSaveEditData(cancel)).done(function() {
-                    deferred.resolve();
-                });
-            };
             this.waitForDeferredOperations().done(() => {
-                if(this._saving) {
-                    afterSaveEditData();
+                if(this.isSaving()) {
+                    this._resolveAfterSaveEditDataComplete(deferred);
                     return;
                 }
                 when(this._beforeSaveEditData()).done(cancel => {
                     if(cancel) {
-                        afterSaveEditData(cancel);
+                        this._resolveAfterSaveEditDataComplete(deferred, { cancel });
+                        this._saving = false;
```

Comment:
Why?

Reply:
I think that _saving = true and _saving = false should be in one place. But now it is changed in different places.